### PR TITLE
New functionality

### PR DESCRIPTION
**Pull request by OlegPoberegets**
## video-quality-tools v1.1.0

* **processFrames**:
    
    Added new fields `gopDuration`, `displayAspectRatio`, `width`, `height`, `hasAudioStream` to the result of  _processFrames_ execution .
    
    Add new methods to _processFrames_: `calculateGopDuration`, `calculateDisplayAspectRatio`, `hasAudioFrames`.
    
* **FramesMonitor**

    FramesMonitor fetches video and audio frames from the stream now.
    
    Added `width` and `height` info to video frames.

### CODE REVIEW

File: src/processFrames.js [hunk 1]
```diff
@@ -4,40 +4,105 @@ const _ = require('lodash');
...
+
+const WIDESCREEN_AR_COEFFICIENT = 2.33;
+const WIDESCREEN_AR = '21:9';
+
 function processFrames(frames) {
     if (!Array.isArray(frames)) {
         throw new TypeError('process method is supposed to accept an array of frames.');
     }
 
+    const audioFrames            = processFrames.filterAudioFrames(frames);
```
Comment (WoZ) [inline src/processFrames.js]: You don't use `audioFrames` array, so you may replace `filterAudioFrames` function with smth like `hasAudioFrames` and use [`Array.some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) method. It's much more efficient.

File: src/processFrames.js [hunk 2]
```diff
@@ -185,6 +250,48 @@ function calculateFps(gops) {
...
+
+    if (Math.abs(arCoefficient - UNIVISIUM_AR_COEFFICIENT) <= AR_CALCULATION_PRECISION) {
+        return UNIVISIUM_AR;
+    }
+
+    if (Math.abs(arCoefficient - WIDESCREEN_AR_COEFFICIENT) <= AR_CALCULATION_PRECISION) {
+        return WIDESCREEN_AR;
+    }
+
+    return `${width}:${height}`;
```
Comment (WoZ) [inline src/processFrames.js]: I think that we need to calculate [GCD](https://en.wikipedia.org/wiki/Greatest_common_divisor). It's common technique in ffmpeg.

**Comment by WoZ** [comment]
You use AR_CALCULATION_PRECISION to normalize result of division but you don't have tests for that.

File: README.md [hunk 3]
```diff
@@ -372,9 +393,10 @@ If there are more than 2 key frames, `processFrames` uses full GOPs to track fps
 in the last GOP that was not finished. It's important to remember the `remainedFrames` output and push a new frame to
 the `remainedFrames` array when it arrives.
 
-For the full GOPs `processFrames` calculates min/max/mean values of bitrates (in kbit/s) and framerates and returns
-them in `payload` field. The result of the check for the similarity of GOP structures for the collected GOPs is returned in
-`areAllGopsIdentical` field.
+For the full GOPs `processFrames` calculates min/max/mean values of bitrates (in kbit/s), framerates and GOP duration 
+(in seconds) and returns them in `payload` field. The result of the check for the similarity of GOP structures for 
+the collected GOPs is returned in `areAllGopsIdentical` field. Fields `width`, `height` and `aspectRatio` adjust on
```
[the remark this answers is not in the document]
Comment (OlegPoberegets) [reply README.md]: \+

File: src/FramesMonitor.js [hunk 4]
```diff
@@ -269,11 +269,9 @@ class FramesMonitor extends EventEmitter {
                     '-hide_banner',
                     '-v',
                     errorLevel,
-                    '-select_streams',
-                    'v:0',
                     '-show_frames',
                     '-show_entries',
-                    'frame=pkt_size,pkt_pts_time,media_type,pict_type,key_frame',
+                    'frame=pkt_size,pkt_pts_time,media_type,pict_type,key_frame,width,height',
```
[the remark this answers is not in the document]
Comment (OlegPoberegets) [reply src/FramesMonitor.js]: Yes, of course.
Here is the example of  the audio frame:

[FRAME]
media_type=audio
key_frame=1
pkt_pts_time=N/A
pkt_size=20
[/FRAME]

File: src/processFrames.js [hunk 5]
```diff
@@ -4,40 +4,105 @@ const _ = require('lodash');
...
 
-    const areAllGopsIdentical = processFrames.areAllGopsIdentical(gops);
-    const bitrate             = processFrames.calculateBitrate(gops);
-    const fps                 = processFrames.calculateFps(gops);
+    let areAllGopsIdentical = true;
+    const hasAudioStream = audioFrames.length > 0;
+    const baseGopSize = gops[0].frames.length;
+    const bitrates = [];
+    const fpsList = [];
+    const gopsDurations = [];
```
[the remark this answers is not in the document]
Comment (OlegPoberegets) [reply src/processFrames.js]: \+ gopDurations

File: src/processFrames.js (hunk 1 above)
Comment (OlegPoberegets) [reply src/processFrames.js]: good remark

File: src/processFrames.js (hunk 2 above)
Comment (OlegPoberegets) [reply src/processFrames.js]: +

File: src/FramesMonitor.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (WoZ) [reply src/FramesMonitor.js]: Git it.

**Pull request merged**